apps/crn-server/src/handlers/webhooks/webhook-contentful.ts

```diff
@@ -0,0 +1,86 @@
+import 'source-map-support/register';
+import Boom from '@hapi/boom';
+import { EventBridge } from 'aws-sdk';
+import { APIGatewayEvent, Handler } from 'aws-lambda';
+import { WebhookDetailType } from '@asap-hub/model';
+import { framework as lambda } from '@asap-hub/services-common';
+import { ContentfulWebhookPayload } from '@asap-hub/contentful';
+import { sentryWrapper } from '../../utils/sentry-wrapper';
+import {
+  contentfulWebhookAuthenticationToken,
+  eventBridgeAccessKey,
+  eventBridgeEndpoint,
+  eventBridgeSecret,
+  eventBus,
+  eventSource,
+} from '../../config';
+
+export const contentfulWebhookFactory = (
+  webhookAuthenticationToken: string,
+  eventBridge: EventBridge,
+): lambda.Handler =>
+  lambda.http<Record<string, string>, void>(
+    async (request: lambda.Request<Record<string, string>>) => {
+      if (!validateContentfulRequest(request, webhookAuthenticationToken)) {
+        throw Boom.unauthorized();
+      }
+
+      const detailType = getDetailTypeFromRequest(request);
+
+      await eventBridge
+        .putEvents({
+          Entries: [
+            {
+              EventBusName: eventBus,
+              Source: eventSource,
+              DetailType: detailType,
+              Detail: JSON.stringify(request.payload),
+            },
+          ],
+        })
+        .promise();
+
+      return {
+        statusCode: 200,
+      };
+    },
+  );
+
+const eventBridge = new EventBridge({
+  endpoint: eventBridgeEndpoint,
+  accessKeyId: eventBridgeAccessKey,
+  secretAccessKey: eventBridgeSecret,
+});
+
+export const handler: Handler<APIGatewayEvent> = sentryWrapper(
+  contentfulWebhookFactory(contentfulWebhookAuthenticationToken, eventBridge),
+);
+
+export const validateContentfulRequest = (
+  request: lambda.Request,
+  webhookAuthenticationToken: string,
+): request is lambda.Request<ContentfulWebhookPayload<'teams' | 'news'>> => {
+  if (!request.headers.authorization) {
+    return false;
+  }
+
+  if (request.headers.authorization !== webhookAuthenticationToken) {
+    return false;
+  }
+
+  return true;
+};
+
+const getDetailTypeFromRequest = (
+  request: lambda.Request<ContentfulWebhookPayload<'teams' | 'news'>>,
```

Comment:
Does `teams | news` need to be expanded to include other things? Or possibly just removed completely? I'm not sure I see the value in defining the type as a generic here.